File: 3461-check-if-digits-are-equal-in-string-after-operations-i/3461-check-if-digits-are-equal-in-string-after-operations-i.py
class Solution:
    def hasSameDigits(self, s: str) -> bool:
        while len(s) > 2:
            temp = []
            for i in range(len(s) - 1):
                digit_sum = (int(s[i]) + int(s[i + 1])) % 10
                temp.append(str(digit_sum))
            s = "".join(temp)
        return s[0] == s[1]

File: 3461-check-if-digits-are-equal-in-string-after-operations-i/3461-check-if-digits-are-equal-in-string-after-operations-i.java
class Solution {
    public boolean hasSameDigits(String s) {
        while(s.length()>2){
            String str="";
            for(int i=0; i<s.length()-1; i++){
                str+=(char)((((s.charAt(i+1)-'0')+(s.charAt(i)-'0'))%10)+'0');
            }
            s=str;
        }
        return s.charAt(0)==s.charAt(1);
    }
}